cpp/tests/transform/row_bit_count_test.cu

```diff
@@ -0,0 +1,583 @@
+
+  cudf::test::strings_column_wrapper col(strings.begin(), strings.end());
+
+  table_view t({col});
+  auto result = cudf::row_bit_count(t);
+
+  // expect 1 offset (4 bytes) + length of string per row
+  auto size_iter = cudf::detail::make_counting_transform_iterator(0, [&strings](int i) {
+    return (static_cast<size_type>(strings[i].size()) + sizeof(offset_type)) * CHAR_BIT;
+  });
+  cudf::test::fixed_width_column_wrapper<size_type> expected(size_iter, size_iter + strings.size());
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, *result);
+}
+
+TEST_F(RowBitCount, StringsWithNulls)
+{
+  // clang-format off
+  std::vector<std::string> strings { "abc", "def", "", "z", "bananas", "warp", "", "zing" };
+  std::vector<bool>        valids  {  1,     0,    0,  1,   0,         1,      1,  1 };
+  // clang-format on
+
+  cudf::test::strings_column_wrapper col(strings.begin(), strings.end(), valids.begin());
+
+  table_view t({col});
+  auto result = cudf::row_bit_count(t);
+
+  // expect 1 offset (4 bytes) + (length of string, or 0 if null) + 1 validity bit per row
+  auto size_iter = cudf::detail::make_counting_transform_iterator(0, [&strings, &valids](int i) {
+    return ((static_cast<size_type>(valids[i] ? strings[i].size() : 0) + sizeof(offset_type)) *
+            CHAR_BIT) +
+           1;
+  });
+  cudf::test::fixed_width_column_wrapper<size_type> expected(size_iter, size_iter + strings.size());
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, *result);
+}
+
+std::pair<std::unique_ptr<column>, std::unique_ptr<column>> build_struct_column()
+{
+  std::vector<bool> struct_validity{0, 1, 1, 1, 1, 0};
+  std::vector<std::string> strings{"abc", "def", "", "z", "bananas", "warp"};
+
+  cudf::test::fixed_width_column_wrapper<float> col0{0, 1, 2, 3, 4, 5};
+  cudf::test::fixed_width_column_wrapper<int16_t> col1{{8, 9, 10, 11, 12, 13}, {1, 0, 1, 1, 1, 1}};
+  cudf::test::strings_column_wrapper col2(strings.begin(), strings.end());
+
+  // creating a struct column will cause all child columns to be promoted to have validity
+  cudf::test::structs_column_wrapper struct_col({col0, col1, col2}, struct_validity);
+
+  // expect (1 offset (4 bytes) + (length of string if row is valid) + 1 validity bit) +
+  //        (1 float + 1 validity bit) +
+  //        (1 int16_t + 1 validity bit) +
+  //        (1 validity bit)
+  auto size_iter =
+    cudf::detail::make_counting_transform_iterator(0, [&strings, &struct_validity](int i) {
+      return (sizeof(float) * CHAR_BIT) + 1 + (sizeof(int16_t) * CHAR_BIT) + 1 +
+             (static_cast<size_type>(strings[i].size()) * CHAR_BIT) +
+             (sizeof(offset_type) * CHAR_BIT) + 1 + 1;
+    });
+  cudf::test::fixed_width_column_wrapper<size_type> expected_sizes(size_iter,
+                                                                   size_iter + strings.size());
+
+  return {struct_col.release(), expected_sizes.release()};
+}
+
+TEST_F(RowBitCount, Structs)
+{
+  // no nulls
+  {
```

Comment:
Maybe make these 3 separate `TEST_F` functions?
```
TEST_F(RowBitCount, Structs)
TEST_F(RowBitCount, StructsNulls)
TEST_F(RowBitCount, StructsNoNulls)
```

Reply:
I went with

```
TEST_F(RowBitCount, StructsNulls)
TEST_F(RowBitCount, StructsNoNulls)
TEST_F(RowBitCount, StructsNested)
```
"Nested" was more descriptive of that particular case.